fuel/datasets/hdf5.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import tables
+
+from fuel.datasets import Dataset
+
+logger = logging.getLogger(__name__)
+
+
+class Hdf5Dataset(Dataset):
+    """An HDF5 dataset
+
+    Parameters
+    ----------
+    sources : tuple of strings
+        Sources which the dataset returns
+    start : int
+        Start index
+    stop : int
+        Stop index
+    data_node : str
+        Parent data node in HDF5 file
+    sources_in_file : tuple of strings
+        Names of nodes in HDF5 file which contain sources. Should the same
+        length as `sources`.
+        Optional, if not set will be equal to `sources`.
+    """
+    def __init__(self, sources, start, stop, path, data_node='Data',
+                 sources_in_file=None):
+        if sources_in_file is None:
+            sources_in_file = sources
+        self.sources_in_file = sources_in_file
+        self.provides_sources = sources
+        self.path = path
+        self.data_node = data_node
+        self.start = start
+        self.stop = stop
+        self.num_examples = self.stop - self.start
+        self.nodes = None
+        self.open_file(self.path)
+        super(Hdf5Dataset, self).__init__(self.provides_sources)
+
+    def open_file(self, path):
+        try:
+            h5file = tables.open_file(path, mode="r")
+            node = h5file.getNode('/', self.data_node)
+
+            self.nodes = [getattr(node, source) for source in self.sources_in_file]
+        except IOError:
+            logger.error('Failed to open HDF5 file, try to call open_file'
```

Comment:
The problem with using `logger.error` is that it will crash if there is no root handler defined in Python 2 (and by default, there is no root handler). We could add a `NullHandler` by default, but that means that the error will pass silently in Python 2 if the user didn't configure a handler. We don't really want to call `logging.basicConfig()` because we're a framework, and client applications might want to define their own root handlers, and it would be bad practice setting them ourself.

tl;dr Logging an error is not a very good way to go. It's better to raise the error, but for that you need to delay loading the file, which is what lazy properties are for.